Create Vite example (#1)
CesiumGS/cesium-vite-example: README.md

# cesium-vite-example

A minimal recommended setup for an applications using [Cesium](https://cesium.com) with [Vite](https://vitejs.dev/).

> If you are using Webpack instead of Vite check out our [`cesium-webpack-example`](https://github.com/CesiumGS/cesium-webpack-example) repo.

Reviewer: ```suggestion
If you are using Webpack instead of Vite, check out our [`cesium-webpack-example`](https://github.com/CesiumGS/cesium-webpack-example) repo.
```

> ## Vue/React/Svelte/etc support

Reviewer: Maybe just title this `UI framework support` and clarify in the text below.

> This example was created to be the lowest common denominator in the Vite ecosystem. The same configuration has been tested with other UI frameworks in Vite with a small modification of adding the relavent plugin and should work the same regardless. If you run into framework specific problems please [open an issue](https://github.com/CesiumGS/cesium-vite-example/issues/new) and we can try and address it.

Reviewer: A few thingss:

1. I would explicitly list which UI frameworks this has been tested with and the name of the plugin needed for each.
2. typo, it should be relevant not relavent
3. I would omit the "and we can try and address it" at the end, it's not really adding anything.

Author: I don't know all the plugins it adds by default for the other frameworks without going through and testing them all but I added a little bit here to make it more general/clear

## Running this application

```sh
npm install
npm run dev
# for the built version
npm run build
npm run preview
```

Navigate to `localhost:5174`. For the built version navigate to `localhost:4174`

## Available scripts

> - `npm run eslint`, `npm run prettier`, `npm run prettier-check` - Lint this project to maintain code style

Reviewer: I would split this into multiple lines since each of these are for slightly different purposes (linting, automatic code formatting, and checking the formatting without modifying files)

- `npm run dev` - Starts the Vite development server server at `localhost:5174`
- `npm run build` - Runs the Vite build
- `npm run preview` - Starts a local preview of the production build using [`vite preview`](https://vitejs.dev/guide/cli.html#vite-preview)

## Requiring Cesium in your application

We recommend [importing named exports](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import) from the Cesium ES module, via the `import` keyword.

### Import named modules from Cesium

```js
import { Color } from "cesium";
var c = Color.fromRandom();
```

### Import Cesium static asset files

```js
import "cesium/Build/Cesium/Widgets/widgets.css";

> ```

Reviewer: Is this a web-framework specific thing that imports a CSS file from JavaScript code? Not familiar with this use of `import`.

Either way, maybe a line to explain where this would be used would be helpful.

Author: This is a general "bundler" style thing. Rollup and esbuild (or webpack elsewhere) is configured to know how to process and include `*.css` files or other style files. That's some of the benefit of bundlers. You can do the same with images and "import" them into JS files.

I changed this a bit to hopefully explain a little better

Reviewer 2: I think users familiar with Vite should be familiar with this style of import.

# cesium-vite-example

A minimal recommended setup for an applications using [Cesium](https://cesium.com) with [Vite](https://vitejs.dev/).

If you are using Webpack instead of Vite, check out our [`cesium-webpack-example`](https://github.com/CesiumGS/cesium-webpack-example) repo.

## UI framework support

> This example was created to be the lowest common denominator in the Vite ecosystem and targets Vanilla JS. The same configuration has been tested with other UI frameworks in Vite (like Vue) with a small modification of adding the relevent plugin and should work the same regardless. If you run into framework specific problems please [open an issue](https://github.com/CesiumGS/cesium-vite-example/issues/new).

Reviewer 2: ```suggestion
This example was created to be the lowest common denominator in the Vite ecosystem and targets Vanilla JS. The same configuration has been tested with other UI frameworks in Vite (like Vue) by adding the relevant plugin. If you run into framework specific problems please [open an issue](https://github.com/CesiumGS/cesium-vite-example/issues/new).
```

> If you create a new Vite project with [`create-vite`](https://vitejs.dev/guide/#scaffolding-your-first-vite-project) you should be able to combine the `plugins` that it adds in `vite.config.js` with the ones in this example configuration and it should work.

Reviewer 2: ```suggestion
If you create a new Vite project with [`create-vite`](https://vitejs.dev/guide/#scaffolding-your-first-vite-project) you can combine the `plugins` that it adds in `vite.config.js` with the ones in this example configuration.
```

## Running this application

```sh
npm install
npm run dev

> # for the built version

Reviewer 2: Should this be a separate code block for clarity?

npm run build
npm run preview
```

Navigate to `localhost:5173`. For the built version navigate to `localhost:4173`

## Available scripts

- `npm run eslint` - Lint this project
- `npm run prettier` - Format all the code to a consistant style
- `npm run prettier-check` - Check the format of code but do not change it
- `npm run dev` - Starts the Vite development server server at `localhost:5173`

> - `npm run build` - Runs the Vite build

Reviewer 2: Just to clarify- This is a production build, not a dev build, correct?
